azimuth/utils/ml/perturbation_functions.py

```diff
@@ -138,7 +142,7 @@ def remove_or_add_final_punctuation(
         perturbation_type = PerturbationType.Deletion
     # Replace with specified punctuation if last character is another punctuation sign.
     elif any([p in last_char for p in [".", "!", "?", ","]]):
```

Comment:
I know this is not new, but... Am I crazy or is that equivalent to this?
```suggestion
    elif last_char in [".", "!", "?", ","]:
```

Reply:
Yes...I think you're right. I think that if `last_char` was defined as more than one character, the logic would technically be different, but since `last_char` is just the last, single character, your suggestion is much much cleaner (and easier to read!). Not my code :) but happy to make the change!